nservicebus/hosting/service-fabric-hosting/hosting-gateway.md

```diff
@@ -0,0 +1,27 @@
+---
+title: Hosting Gateway with Service Fabric
+related:
+ - nservicebus/service-fabric
+reviewed: 2017-03-30
+---
+
+When adopting Service Fabric, it's not uncommon that the Service Fabric hosted endpoints need to interact with endpoints outside of the cluster. This can get tricky especially when the endpoints inside Service Fabric are stateful. When integrating using client side distribution, or when using the Service Fabric built in [reverse proxy](https://docs.microsoft.com/nl-nl/azure/service-fabric/service-fabric-reverseproxy) to expose the endpoints as web services, then the partition information needs to be provided by the consumer, which is often not desired.
+
+Alternatively the [nservicebus gateway](/nservicebus/gateway/) can be leveraged as an intermediary to solve this problem. It provides reliable request reply semantics, with deduplication, between sites.
+
+To host the NServiceBus Gateway in an endpoint deployed to Service Fabric, the following has to be taken into account:
+
+1. Host the gateway as a [stateless service](/nservicebus/hosting/service-fabric/hosting/#stateful service), and use [partition aware routing](/samples/azure/azure-service-fabric-routing/) within to forward messages to other parts of the cluster.
+2. Because the service instance are hosted behind a load balancer, it is required to configure the gateway channel address to use a URL with wildcard. The reason is that the public IP address is the one from the cluster load balancer and will therefore differ from the one that the gateway communication listener will effectively be listening on. 
+
+Snippet: configureWildcardGatewayChannel
+
+3. Open the correct port for the gateway on the cluster load balancer by specifying an `Endpoint` entry in the service manifest.
+
+Snippet: serviceManifestEndpoint
+
+4. Add a second communication listener that returns the gateway address using the local FQDN instead of the + sign so that the service fabric instance is aware of the local address.
```

Comment:
For me this is a confusing. AFAIK when a `ICommunicationListener` returns a url SF does at least two things:
 * It configures `urlacl` for that url for process running the service - this is also done by by `Gateway` itself
 * It registers the url in `NamingService` so that it's discoverable inside the cluster - this is not needed in the approach we guide users to use - `Gateway` behind Load Balancer and deployed on all the nodes.

In other words for me it's not clear what `service fabric instance is aware of the local address` means and secondly if this is even needed.